libres/tests/analysis/ies/test_ies_enkf_main.cpp

```diff
@@ -38,51 +35,142 @@ TEST_CASE("ies_enkf_linalg_extract_active", "[analysis]") {
-    ies_enkf_linalg_extract_active(data, E_ens_deactivate, stdout, false);
-
-    REQUIRE(matrix_iget(E_ens_deactivate, 0, 0) == 1.0);
-    REQUIRE(matrix_iget(E_ens_deactivate, 1, 0) == 2.0);
-    REQUIRE(matrix_iget(E_ens_deactivate, 2, 0) == 3.0);
-    REQUIRE(matrix_iget(E_ens_deactivate, 0, 1) == 0.0);
-    REQUIRE(matrix_iget(E_ens_deactivate, 1, 1) == 0.0);
-    REQUIRE(matrix_iget(E_ens_deactivate, 2, 1) == 0.0);
+    ies::enkf_data_update_ens_mask(data, ens_mask);
+    {
+        matrix_type *E = ies::enkf_alloc_activeE(data);
+        REQUIRE(matrix_iget(E, 0, 0) == 1.0);
+        REQUIRE(matrix_iget(E, 1, 0) == 2.0);
+        REQUIRE(matrix_iget(E, 2, 0) == 3.0);
+        REQUIRE(matrix_get_rows(E) == 3);
+        REQUIRE(matrix_get_columns(E) == 1);
+        matrix_free(E);
+    }
 
     // Test that `ies_enkf_linalg_extract_active` can deactivate an observation
     bool_vector_iset(obs_mask, 1, false);
-    ies_enkf_update_obs_mask(data, obs_mask);
-
-    matrix_type *E_obs_deactivate = matrix_alloc(state_size, ens_size);
-    ies_enkf_linalg_extract_active(data, E_obs_deactivate, stdout, false);
-
-    REQUIRE(matrix_iget(E_obs_deactivate, 0, 0) == 1.0);
-    REQUIRE(matrix_iget(E_obs_deactivate, 1, 0) == 3.0);
-    REQUIRE(matrix_iget(E_obs_deactivate, 2, 0) == 0.0);
-    REQUIRE(matrix_iget(E_obs_deactivate, 0, 1) == 0.0);
-    REQUIRE(matrix_iget(E_obs_deactivate, 1, 1) == 0.0);
-    REQUIRE(matrix_iget(E_obs_deactivate, 2, 1) == 0.0);
-
+    ies::enkf_update_obs_mask(data, obs_mask);
+    {
+        matrix_type *E = ies::enkf_alloc_activeE(data);
+        REQUIRE(matrix_iget(E, 0, 0) == 1.0);
+        REQUIRE(matrix_iget(E, 1, 0) == 3.0);
+        REQUIRE(matrix_get_rows(E) == 2);
+        REQUIRE(matrix_get_columns(E) == 1);
+        matrix_free(E);
+    }
+
+    matrix_free(Ein);
     bool_vector_free(ens_mask);
     bool_vector_free(obs_mask);
-    matrix_free(E);
-    matrix_free(E_ens_deactivate);
-    matrix_free(E_obs_deactivate);
-    ies_enkf_data_free(data);
+    ies::enkf_data_free(data);
     rng_free(rng);
 }
+
+TEST_CASE("ies_enkf_linalg_extract_active_W", "[analysis]") {
+    const int ens_size = 4;
+    const int obs_size = 10;
+    auto *data = static_cast<ies::enkf_data_type *>(ies::enkf_data_alloc());
+    bool_vector_type *ens_mask = bool_vector_alloc(ens_size, true);
+    bool_vector_type *obs_mask = bool_vector_alloc(obs_size, true);
+
+    ies::enkf_config_set_logfile(ies::enkf_data_get_config(data), "log");
+    ies::enkf_data_open_log(data);
+    ies::enkf_init_update(data, ens_mask, obs_mask, nullptr, nullptr, nullptr,
+                          nullptr, nullptr, nullptr);
+    ies::enkf_data_update_ens_mask(data, ens_mask);
+
+    matrix_type *W0 = matrix_alloc(ens_size, ens_size);
+    for (int i = 0; i < ens_size; i++) {
+        for (int j = 0; j < ens_size; j++)
+            matrix_iset(W0, i, j, i * ens_size + j);
+    }
+    ies::enkf_linalg_store_active_W(data, W0);
```

Comment:
Since these are being renamed already, there are a couple more prefixes to these names that I am wondering whether is best omitted. That is `ies::enkf_*`, `ies::linalg_*` and `ies::enkf_linalg_*`. It might be room for a `ies::linalg::` namespace, but it seems that all functions currently with `linalg` in their names are not purely about linear algebra.

Reply:
I can do renames, but I do not fully understand what you request here. 

Must admit I feel this PR has been in the works quite long now and would be happy if it could be merged.

Reply:
I'm sure future PR's will get merged much faster, as we are using this PR to discuss and agree on how we want the code to look, which is quite time-consuming.

Reply:
Maybe you could split the rename into a different PR, then we can speed up the review of the other stuff and look at the renaming later.

Reply:
Of course I can submit renames in a separate PR - that was implicitly what I suggested in me previous comment.

Reply:
[Explicit is better than implicit](https://zen-of-python.info/explicit-is-better-than-implicit.html#2) 😄

Reply:
Sorry, I was being ambiguous again, I was thinking that renaming and introduction of namespace belonged together so that 979d063 should go in the split out PR.

Reply:
Sorry - still not entirely clear. Are now asking me to relegate the new `ies::` namespace to a new PR? Of course I can do that - but this review process feels quite schizophrenic from my end.

Reply:
Sorry about the schizophrenic behavior, but I think its the best course of action. Then we will have discharged that point and can fast track the other parts :)

Reply:
We should probably have considered making a separate PR for the ies namespace in the first place, but at least now we know :)

Reply:
> I can do renames, but I do not fully understand what you request here.
> 
> Must admit I feel this PR has been in the works quite long now and would be happy if it could be merged.

We did not start looking at it before it was marked as "Ready for review", which was a couple of days ago if I remember correctly.
This perhaps add to the feeling that it's taking long to review.